lib/matplotlib/pyplot.py

```diff
@@ -557,6 +564,9 @@ def make_active(event):
         if _INSTALL_FIG_OBSERVER:
             fig.stale_callback = _auto_draw_if_interactive
 
+    if clf and len(figManager.canvas.figure.axes) > 0:
+        figManager.canvas.figure.clear()
```

Comment:
Without committing myself yet as to whether the PR as a whole is a good idea, I think that the clearing should be unconditional, not based on whether there are axes objects.  Otherwise, it is only a partial clearing of the figure.